src/mpid/common/genq/mpidu_genq_shmem_queue.c

```diff
@@ -4,28 +4,48 @@
  */
 
 #include "mpidimpl.h"
-#include "mpidu_genq_shmem_pool.h"
-#include "mpidu_genq_shmem_queue.h"
-#include "mpidu_genqi_shmem_types.h"
 
-int MPIDU_genq_shmem_queue_init(MPIDU_genq_shmem_queue_t queue, MPIDU_genq_shmem_pool_t pool,
-                                int flags)
+int MPIDU_genq_shmem_queues_alloc(int num_of_queues, MPIDU_genq_shmem_queues_t * queues)
```

Comment:
I don't think this function is necessary. It wraps shmem allocation and make it a shmem calloc with fixed element size. 

Can we just expose the `sizeof(MPIDU_genq_shmem_queue_u)` as a macro, so the user can handle shmem allocation without explicitly knowing the `MPIDU_genq_shmem_queue_u` type.